evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -75,20 +75,17 @@ def __init__(self,
 
     @staticmethod
     def _get_cat_cols(X):
-        """Get names of 'object' or 'categorical' columns in the DataFrame."""
-        obj_cols = []
-        for idx, dtype in enumerate(X.dtypes):
-            if dtype == np.object or pd.api.types.is_categorical_dtype(dtype):
-                obj_cols.append(X.columns.values[idx])
-        return obj_cols
+        """Get names of categorical columns in the input Woodwork DataTable."""
+        return list(X.select(include=['category']).columns.keys())
 
     def fit(self, X, y=None):
         top_n = self.parameters['top_n']
         X = _convert_to_woodwork_structure(X)
+        if self.features_to_encode is None:
+            self.features_to_encode = self._get_cat_cols(X)
```

Comment:
Did moving this code change any behavior? Or was it just cleaner this way, but functionally equivalent?

Reply:
It shouldn't change behavior; the reason for moving this is because we want to take advantage of Woodwork so we want to pass in X as a Woodwork structure before converting it later :)